app/lib/methods/getAppActions.ts

```diff
@@ -0,0 +1,126 @@
+		const db = database.active;
+		try {
+			const dbCollection = db.get('app_actions_buttons');
+			const appRecords = await dbCollection.query(Q.where('app_id', appId)).fetch();
+			if (appRecords) {
+				await db.write(async () => {
+					await Promise.all(appRecords.map(app => app.destroyPermanently()));
+				});
+				reduxStore.dispatch(removeAppActionButtonsByAppId(appId));
+			}
+		} catch (e) {
+			log(e);
+		}
+	}
+
+	if (/app\/added|actions\/changed|app\/statusUpdate|app\/updated/.test(eventName)) {
+		try {
+			await getAppActions();
+		} catch (e) {
+			log(e);
+		}
+	}
+}
+
+export const getAppActions = () => {
+	const db = database.active;
+
+	return new Promise<void>(async resolve => {
+		try {
+			const appActionButtons = (await sdk.get('actionButtons', undefined, 'apps')) as unknown as IAppActionButton[];
+
+			if (appActionButtons && appActionButtons.length) {
+				await db.write(async () => {
+					const appActionButtonsCollection = db.get('app_actions_buttons');
+					const allAppActionButtonsRecords = await appActionButtonsCollection.query().fetch();
+
+					const filteredAppActionButtonsToCreate = appActionButtons.filter(
+						(i1: IAppActionButton) =>
+							!allAppActionButtonsRecords.find(i2 => `${i1.appId}/${i1.actionId}` === `${i2.appId}/${i2.actionId}`)
+					);
+
+					const filteredAppActionButtonsToUpdate = allAppActionButtonsRecords.filter(i1 =>
+						appActionButtons.find((i2: IAppActionButton) => `${i1.appId}/${i1.actionId}` === `${i2.appId}/${i2.actionId}`)
+					);
+
+					const filteredAppActionButtonsToDelete = allAppActionButtonsRecords.filter(
+						i1 =>
+							!filteredAppActionButtonsToCreate.find(
+								(i2: IAppActionButton) => `${i1.appId}/${i1.actionId}` === `${i2.appId}/${i2.actionId}`
+							) && !filteredAppActionButtonsToUpdate.find(i2 => `${i1.appId}/${i1.actionId}` === `${i2.appId}/${i2.actionId}`)
+					);
+
+					const appActionButtonsToCreate = filteredAppActionButtonsToCreate.map((action: IAppActionButton) =>
+						appActionButtonsCollection.prepareCreate(
+							protectedFunction((s: TAppActionButtonModel) => {
+								s._raw = sanitizedRaw({ id: `${action.appId}/${action.actionId}` }, appActionButtonsCollection.schema);
+								Object.assign(s, action);
+							})
+						)
+					);
+
+					const appActionButtonsToUpdate = filteredAppActionButtonsToUpdate.map(action => {
+						const newAction = appActionButtons.find(
+							(s: IAppActionButton) => `${s.appId}/${s.actionId}` === `${action.appId}/${action.actionId}`
+						);
+						return action.prepareUpdate(
+							protectedFunction((s: TAppActionButtonModel) => {
+								Object.assign(s, newAction);
+							})
+						);
+					});
+
+					const appActionButtonsToDelete = filteredAppActionButtonsToDelete.map(action => action.prepareDestroyPermanently());
+
+					const allRecords = [...appActionButtonsToCreate, ...appActionButtonsToUpdate, ...appActionButtonsToDelete];
+
+					try {
+						await db.batch(...allRecords);
```

Comment:
It should be possible to send the whole array now https://github.com/RocketChat/Rocket.Chat.ReactNative/pull/5987
```suggestion
						await db.batch(allRecords);
```